### PR TITLE
Add release note for 1.2.1

### PR DESCRIPTION
**Pull request by pubudu91**

### CODE REVIEW

File: release-notes-1.2.1.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+- jBallerina 1.2.0 but switched to a previous version: `$ ballerina dist pull jballerina-1.2.1`
+- a jBallerina version below 1.1.0: install via the [installers](https://ballerina.io/downloads/)
+
+**For new users:**
+
+If you have not installed jBallerina, then download the [installers](https://ballerina.io/downloads/) to install.
+
+## Language
+- Deprecation support for annotations, function parameters and object fields
```
Comment (praneesha) [inline release-notes-1.2.1.md:23]:
```suggestion
- Deprecation support for annotations, function parameters, and object fields
```

File: release-notes-1.2.1.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- Runtime validations added to XML attributes map.
```
Comment (praneesha) [inline release-notes-1.2.1.md:24]:
```suggestion
- Runtime validations added to the XML attributes map.
```

File: release-notes-1.2.1.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- Support for map to record assignment
```
Comment (praneesha) [inline release-notes-1.2.1.md:25]:
```suggestion
- Support for a map to record the assignments
```

File: release-notes-1.2.1.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Standard Library
+- Support for Avro key serialization/deserialization in `ballerina/kafka` module
```
Comment (praneesha) [inline release-notes-1.2.1.md:28]:
```suggestion
- Support for Avro key serialization/deserialization in the `ballerina/kafka` module
```

File: release-notes-1.2.1.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+- Automatic failover support for WebSocket
+- Provide the base64 encoded end user certificate along with the mutual SSL status.
```
Comment (praneesha) [inline release-notes-1.2.1.md:30]:
```suggestion
- Provide the base64-encoded end user certificate along with the mutual SSL status
```

File: release-notes-1.2.1.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	
+## Dev tools
+- Support for mocking object methods in Testerina
+- Support for subtypes in Java for the bindgen tool
+- Support for formatting for streams
```
Comment (praneesha) [inline release-notes-1.2.1.md:35]:
```suggestion
- Support for formatting of streams
```

File: release-notes-1.2.1.md (hunk 3 above)
Comment (pubudu91) [reply release-notes-1.2.1.md:25]: This is actually correct. This refers to assigning map types to record types

**Pull request merged**